File: SpinnerViewExeample/app/src/main/java/com/meteoro/spinnerviewexeample/MainActivity.java
package com.meteoro.spinnerviewexeample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Spinner;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    public Spinner spinner;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        spinner = findViewById(R.id.spinner);

        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(
                MainActivity.this,
                R.array.names_array,
                android.R.layout.simple_spinner_item
        );

        spinner.setAdapter(adapter);

        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int pos, long id) {
                String nome = (String) adapterView.getItemAtPosition(pos);
                Log.d("main", nome);
                Toast.makeText(MainActivity.this, "Selecionou: " + nome, Toast.LENGTH_LONG).show();
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {

            }
        });
    }
}

File: DogAPI/app/src/main/java/com/meteoro/dogapi/MainActivity.kt
package com.meteoro.dogapi

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import com.squareup.picasso.Picasso
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    private lateinit var btnRandomDog: Button
    private lateinit var imageDog: ImageView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageDog = findViewById(R.id.image_dog)
        btnRandomDog = findViewById(R.id.btn_random_dog)
        btnRandomDog.setOnClickListener {
            makeRandomHttpAccess()
        }
    }

    private fun makeRandomHttpAccess() {
        val retrofit: Retrofit = Retrofit.Builder()
            .baseUrl("https://dog.ceo/api/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

        val dogService: DogService = retrofit.create(DogService::class.java)

        val call = dogService.randomDog().enqueue(object : Callback<Dog> {
            override fun onResponse(call: Call<Dog>, response: Response<Dog>) {
                val randomDog = response.body()!!

                Log.d("MainActivity", randomDog.message)
                Log.d("MainActivity", randomDog.status)

                Toast.makeText(this@MainActivity, "${randomDog.status}", Toast.LENGTH_LONG).show()

                Picasso.get()
                    .load(randomDog.message)
                    .into(imageDog)
            }

            override fun onFailure(call: Call<Dog>, error: Throwable) {
                Log.e("MainActivity", "Deu erro -> $error")
            }
        })

        Log.d("MainActivity", "Antes do retorno")
    }
}
